net-dns/yaddns/yaddns-1.1.recipe

```diff
@@ -0,0 +1,74 @@
+SUMMARY="A small dynamic DNS client"
+DESCRIPTION="Yaddns is a small dynamic dns client with multiple servers \
+support and a high flexibility."
+HOMEPAGE="https://yaddns.github.io/"
+COPYRIGHT="2015 Anthony Viallard and Raphael Huck"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/yaddns/yaddns/archive/yaddns-$portVersion.tar.gz"
+CHECKSUM_SHA256="23b527b7f71e1449746f1e39cbc3a074d58bd261a33399fb4e2d90eb63149059"
+SOURCE_DIR="yaddns-yaddns-$portVersion"
+PATCHES="yaddns-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
+
+PROVIDES="
+	yaddns$secondaryArchSuffix = $portVersion
+	cmd:yaddns$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:autom4te
+	cmd:automake
+	cmd:autoreconf
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	"
+
+defineDebugInfoPackage yaddns$secondaryArchSuffix \
+	$commandBinDir/yaddns
+
+PATCH()
+{
+	sed -i -e "s|/etc\(/yaddns\.conf\)|`finddir B_SYSTEM_SETTINGS_DIRECTORY`\1|;" \
+		doc/yaddns.1
+}
+
+BUILD()
+{
+	./autogen.sh
+	LIBS="-Wl,--as-needed -lbnetapi -lnetwork" runConfigure \
+		--omit-dirs binDir ./configure --bindir $commandBinDir
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+}
+
+TEST()
+{
+	make check
```

Comment:
```diff
 INSTALL()
{
	make install
+	install -d $docDir
+	install AUTHORS $docDir
+	install etc/yaddns.conf $docDir/yaddns.conf.example
 }

 TEST()
 {
-	make check
+	LIBS="-Wl,--as-needed -lbnetapi -lnetwork" make check
 }
```
* INSTALL: @alaviss, shall we add the example config?
* TEST: This change is required, but, unfortunately, it is not enough to be able to run `haikuporter --test yaddns_x86`:
```
../src/util.o: In function `util_getifaddr':
util.c:(.text+0x22c): undefined reference to `getifaddr'
```
@sfanxiang, in case you would like to try and patch the sources used in `TEST()`, we can keep the PR open until you have fixed it.
But, if you prefer, we can merge the PR now and postpone the fix.
(BTW, you don't even need to do the fix, but this would be great if you did so!)

Reply:
@fbrosson OK, I'll look into `TEST()`, if you can wait for one or two days.

Reply:
Cool, great!
Of course we can wait a few days. We can even wait longuer. You can consider that there is no deadline for fixing yaddns's TEST.
BTW, the task you claimed about 90 minutes ago is a great choice!

So please don't hesitate to work on your new task (and maybe more others) until the end of the GCI, and come back to yaddns's TEST later.
(This is only a suggestion, of course. You can chose to fix yaddns's TEST when it wish.)

(BTW, we have several open PR's from other contributors, and this does not have any impact on the fact that the task was validated. After all, your recipe is already working perfectly.)

Reply:
@fbrosson Sure, but I would recommend the snippet [here][0] instead. It's clean, and have explanation for each options included

[0]: https://github.com/yaddns/yaddns/wiki/How-to-use-yaddns#yaddnsconf

Reply:
@fbrosson I just fixed `make check`. All tests are green here on x86_64. Could you please test it on x86 as well?

Reply:
Build is ok on x86_gcc2 (for x86), but test fails here (5 out of 5) ...

Reply:
@Begasus Thanks for testing. Could you please post the test message here (or link to a gist)?

Reply:
https://pastebin.com/WmC4ujxZ

Reply:
@Begasus It seems that `tests/test-suite.log` contains the log data. Could you please upload this file as well?

Reply:
Same results for me: All 5 tests pass on x86_64, and all fail on x86_gcc2:

Test | x86 2nd arch | x86_64
--- | --- | ---
check_request | FAIL | PASS
check_cfgstr | FAIL | PASS
check_config | FAIL | PASS
check_account | FAIL | PASS
check_util | FAIL | PASS

BTW, thanks, @sfanxiang, for having fixed `make check` ! And @Begasus for testing on x86 2nd arch.

Now, a political question: shall we merge this with x86 second arch marked as tested, or marked as untested?
(I think we can mark it as tested. The runtime builds fine and does not crash. Maybe it works fine, and the tests fail for some other reason. So if we mark it as untested we'll never get any report that it either works or fails. What do you think?
If we ever get reports that it does not work on x86 we can always change `x86` to `?x86`.

@alaviss, regarding the snippet conf, I'm not comfortable with providing a config which makes yaddns connect to a server. IMO the only acceptable config would be one which would connect to localhost :-) But even this could be a problem. Imagine someone launches yaddns on a server running a webserver. We can't use localhost, after all. We can't use example.org, because we can't assume the user would be OK with example.org knowing that that user is running yaddns. Last but not least, yaddns does not use encryption, so if he customizes his conf with his credentials but forgets to change the server, he will be disclosing secret info to a 3rd party server. BTW, we have a task for creating a recipe for ndyndns, and that tool has support for SSL :)

Reply:
@fbrosson or you could wait a bit and let @sfanxiang dig into x86 problems.

Regarding the snippet conf, if you feel concerned about user error, let them make one themselves with those man pages is fine to me

Reply:
Of course I can wait! It's always better to fix things when possible, so we can keep this PR open until everyone is OK with the merge.

BTW, there is no urgency and we are all having fun. Why would we choose to merge now if we can continue having fun, and, above all, improve the recipe?

@sfanxiang, you can even give yaddns a rest if you wish. I'll only merge when you tell me it's OK to merge. It's really great to have you all, creating great recipes!

Reply:
Sorry @alaviss, I think I had not understood your suggestion about the snippet.
I thought you were suggesting that we install that snippet as the default config.
But I guess you were suggesing to use that snippet as the example config.
I'm perfecly OK with that. But this means we would have to add it to the `additional-files` directory.
And I don't know if the other key contributors would be OK with this. Because anyone can access the project's wiki online, so providing a snippet that is available online might not get the green lights from the other key contributors.
But you can discuss this.
I think we don't really need an alternate snippet. But I'm OK if you and @sfanxiang would like to add one.